internal/runners/initialize/init.go

```diff
@@ -33,19 +32,36 @@ type RunParams struct {
 
 // Initialize stores scope-related dependencies.
 type Initialize struct {
-	out output.Outputer
+	config projectfile.ConfigGetter
+	out    output.Outputer
 }
 
 type primeable interface {
+	primer.Configurer
 	primer.Outputer
 }
 
 // New returns a prepared ptr to Initialize instance.
 func New(prime primeable) *Initialize {
-	return &Initialize{prime.Output()}
+	return &Initialize{prime.Config(), prime.Output()}
 }
 
-func sanitize(params *RunParams) error {
+func sanitize(params *RunParams, config projectfile.ConfigGetter) error {
+	// Try to infer the language from the project currently in use (i.e. `state use show`).
+	// Error handling is not necessary because it's an input error to not include a language to
+	// `state init`. We're just trying to infer one as a convenience to the user.
+	if params.Language == "" {
+		if projectDir := config.GetString(constants.GlobalDefaultPrefname); projectDir != "" {
+			if proj, err := project.FromPath(projectDir); err == nil {
+				if commitID := proj.CommitUUID(); commitID != "" {
+					if lang, err := model.FetchLanguageForCommit(commitID); err == nil {
+						params.Language = fmt.Sprintf("%s@%s", lang.Name, lang.Version)
+					}
+				}
+			}
+		}
+	}
```

Comment:
nit: is there any way we can reduce the nesting? This could also be a helper function.